src/SparkVault.sol

```diff
@@ -341,12 +344,20 @@ contract SparkVault is AccessControlEnumerableUpgradeable, UUPSUpgradeable, ISpa
         return _divup(shares * nowChi(), RAY);
     }
 
-    function previewRedeem(uint256 shares) external view returns (uint256) {
-        return convertToAssets(shares);
+    function previewRedeem(uint256 shares) external view returns (uint256 amount) {
+        amount = convertToAssets(shares);
```

Comment:
Lets update all the preview functions to be public, and then call them from within `deposit` `mint` `withdraw` `redeem`

Reply:
as discussed we are not doing this due to difference in calculation